### PR TITLE
GEODE-9209: Complete test in DistributionConfigJUnitTest

### PR DESCRIPTION
**Pull request by alb3rtobr**
`testCheckerChecksValidAttribute` in `DistributionConfigJUnitTest` has a "TODO" comment about a missing check in the test.

I have also changed JUnit asserts by AssertJ asserts, which are more descriptives in case of error.

### CODE REVIEW

**Comment by onichols-pivotal** [comment]
PR checks in this PR are now a week old (and don't include DistributedTestOpenJDK8).  If possible @alb3rtobr could you please push an empty commit?

**Comment by alb3rtobr** [comment]
Kind reminder, review by code owners still pending

File: geode-core/src/test/java/org/apache/geode/distributed/internal/DistributionConfigJUnitTest.java [hunk 1]
```diff
@@ -162,17 +159,17 @@ public void testAttributeDesc() {
 
   @Test
   public void sameCount() {
-    assertEquals(attributes.size(), setters.size());
-    assertEquals(setters.size(), getters.size());
+    assertThat(setters).hasSize(attributes.size());
+    assertThat(getters).hasSize(setters.size());
```
Comment (kirklund) [inline geode-core/src/test/java/org/apache/geode/distributed/internal/DistributionConfigJUnitTest.java:163]: There is a `hasSameSizeAs` assertion that  is cleaner and provides better failure messages:
```
assertThat(setters).hasSameSizeAs(attributes);
assertThat(getters).hasSameSizeAs(setters);
```

File: geode-core/src/test/java/org/apache/geode/distributed/internal/DistributionConfigJUnitTest.java [hunk 2]
```diff
@@ -162,17 +159,17 @@ public void testAttributeDesc() {
...
 
   @Test
   public void everyAttrHasValidSetter() {
     for (String attr : attributes.keySet()) {
       Method setter = setters.get(attr);
-      assertNotNull(attr + " should have a setter", setter);
-      assertTrue(setter.getName().startsWith("set"));
-      assertEquals(setter.getParameterCount(), 1);
+      assertThat(setter).as(attr + " should have a setter").isNotNull();
+      assertThat(setter.getName().startsWith("set")).isTrue();
```
Comment (kirklund) [inline geode-core/src/test/java/org/apache/geode/distributed/internal/DistributionConfigJUnitTest.java:171]: A better way to form this assertion with better failure message:
```
assertThat(setter.getName()).startsWith("set");
```

File: geode-core/src/test/java/org/apache/geode/distributed/internal/DistributionConfigJUnitTest.java [hunk 3]
```diff
@@ -235,55 +232,59 @@ public void everyGetterSetterSameNameSameType() {
...
-      assertNotNull("every getter should have a corresponding setter " + attr, setter);
+      assertThat(setter).as("every getter should have a corresponding setter " + attr).isNotNull();
       String setterName = setter.getName();
       String getterName = getter.getName();
-      assertEquals(setterName.substring(setterName.indexOf("set") + 3),
-          getterName.substring(getterName.indexOf("get") + 3));
-      assertEquals(setter.getParameterTypes()[0], getter.getReturnType());
+      assertThat(getterName.substring(getterName.indexOf("get") + 3))
+          .isEqualTo(setterName.substring(setterName.indexOf("set") + 3));
+      assertThat(getter.getReturnType()).isEqualTo(setter.getParameterTypes()[0]);
```
Comment (kirklund) [inline geode-core/src/test/java/org/apache/geode/distributed/internal/DistributionConfigJUnitTest.java:240]: Another String assertion syntax that works better is:
```
assertThat("foobar").contains("foo");
```
I would lean away from using`substring` and `indexOf` within assertions.

File: geode-core/src/test/java/org/apache/geode/distributed/internal/DistributionConfigJUnitTest.java [hunk 4]
```diff
@@ -235,55 +232,59 @@ public void everyGetterSetterSameNameSameType() {
...
   public void testGetAttributeObject() {
-    assertEquals(config.getAttributeObject(LOG_LEVEL), "config");
-    assertEquals(config.getAttributeObject(SECURITY_LOG_LEVEL), "config");
-    assertEquals(config.getAttributeObject(REDUNDANCY_ZONE), "");
-    assertEquals(config.getAttributeObject(ENABLE_CLUSTER_CONFIGURATION).getClass(), Boolean.class);
+    assertThat(config.getAttributeObject(LOG_LEVEL)).isEqualTo("config");
+    assertThat(config.getAttributeObject(SECURITY_LOG_LEVEL)).isEqualTo("config");
+    assertThat(config.getAttributeObject(REDUNDANCY_ZONE)).isEqualTo("");
+    assertThat(config.getAttributeObject(ENABLE_CLUSTER_CONFIGURATION).getClass())
+        .isEqualTo(Boolean.class);
```
Comment (kirklund) [inline geode-core/src/test/java/org/apache/geode/distributed/internal/DistributionConfigJUnitTest.java:271]: Better assertion syntax is:
```
assertThat(config.getAttributeObject(ENABLE_CLUSTER_CONFIGURATION)).isInstanceOf(Boolean.class);
```

File: geode-core/src/test/java/org/apache/geode/distributed/internal/DistributionConfigJUnitTest.java [hunk 5]
```diff
@@ -235,55 +232,59 @@ public void everyGetterSetterSameNameSameType() {
...
   }
 
   @Test
   public void testCheckerChecksValidAttribute() {
     for (String att : checkers.keySet()) {
       System.out.println("att = " + att);
-      assertTrue(attributes.containsKey(att));
+      assertThat(attributes.containsKey(att)).isTrue();
```
Comment (kirklund) [inline geode-core/src/test/java/org/apache/geode/distributed/internal/DistributionConfigJUnitTest.java:278]: Recommend:
```
assertThat(attributes).containsKey(att);
```

**Review by kirklund — CHANGES_REQUESTED** [review]

File: geode-core/src/test/java/org/apache/geode/distributed/internal/DistributionConfigJUnitTest.java (hunk 3 above)
Comment (alb3rtobr) [reply geode-core/src/test/java/org/apache/geode/distributed/internal/DistributionConfigJUnitTest.java]: I have refactored the assertion about the setter and getter names as:
```
String attrNameInGetterSignature = getter.getName().substring(3);
assertThat(setter.getName()).contains(attrNameInGetterSignature);
```
I think now its easier to understand what it is being checked there.

**Comment by alb3rtobr** [comment]
Thanks for the review @kirklund , I have implemented the requested changes

**Review by echobravopapa — COMMENTED** [review]
LGTM, but the PR checks need to run and Kirk asked for some changes

File: geode-core/src/test/java/org/apache/geode/distributed/internal/DistributionConfigJUnitTest.java [hunk 6]
```diff
@@ -138,19 +135,19 @@ public void testGetAttributeNames() {
...
   }
 
   @Test
   public void testAttributeDesc() {
     String[] attNames = AbstractDistributionConfig._getAttNames();
     for (String attName : attNames) {
-      assertTrue("Does not contain description for attribute " + attName,
-          AbstractDistributionConfig.dcAttDescriptions.containsKey(attName));
+      assertThat(AbstractDistributionConfig.dcAttDescriptions.containsKey(attName))
+          .as("Does not contain description for attribute " + attName).isTrue();
```
Comment (kirklund) [inline geode-core/src/test/java/org/apache/geode/distributed/internal/DistributionConfigJUnitTest.java:150]: Many assertions still need further change after using the "Convert Assertions to AssertJ" IntelliJ plugin. Assertions on collections usually require some further help.

Let's change this assertion to:
```
assertThat(AbstractDistributionConfig.dcAttDescriptions).containsKey(attName);
```

**Review by kirklund — CHANGES_REQUESTED** [review]
One collection assertion wasn't fixed up. I'll check back tomorrow to see if it's done and add my approval.

**Review by kirklund — APPROVED** [review]

**Pull request merged**